subsys/caf/modules/power_manager.c

```diff
@@ -77,6 +81,84 @@ static void power_down_counter_abort(void)
 	LOG_DBG("Power down timer aborted");
 }
 
+static int settings_set(const char *key, size_t len_rd, settings_read_cb read_cb, void *cb_arg)
+{
+	if (!strcmp(key, POWER_DOWN_TIMEOUT_CONFIG_KEY)) {
+		ssize_t len = read_cb(cb_arg, &power_down_timeout, sizeof(power_down_timeout));
+
+		if ((len != sizeof(power_down_timeout)) || (len != len_rd)) {
+			LOG_ERR(
+				"Can't read power_down_timeout from storage. Using default: %d [s]",
+				power_down_timeout
+			);
+			return len;
+		}
+		LOG_INF("Power down timeout set to %d [s]", power_down_timeout);
+	}
+
+	if (!strcmp(key, POWER_DOWN_ERROR_TIMEOUT_CONFIG_KEY)) {
+		ssize_t len = read_cb(cb_arg, &power_down_error_timeout, sizeof(power_down_error_timeout));
+
+		if ((len != sizeof(power_down_error_timeout)) || (len != len_rd)) {
+			LOG_ERR(
+				"Can't read power_down_error_timeout from storage. Using default: %d [s]",
+				power_down_error_timeout
+			);
+			return len;
+		}
+		LOG_INF("Power down error timeout set to %d [s]", power_down_error_timeout);
+	}
+
+	return 0;
+}
+
+#if IS_ENABLED(CONFIG_CAF_POWER_MANAGER_RUNTIME_TIMEOUT_SET)
```

Comment:
btw - I would add a separate config option for RUNTIME_TIMEOUT_SAVE. Somebody may want to configure it but may not want it to be stored.